numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -427,7 +427,11 @@ PyArray_ConvertMultiAxis(PyObject *axis_in, int ndim, npy_bool *out_axis_flags)
 NPY_NO_EXPORT int
 PyArray_BoolConverter(PyObject *object, npy_bool *val)
 {
-    if (PyObject_IsTrue(object)) {
+    int istrue = PyObject_IsTrue(object);
+    if (istrue == -1) {
+        return NPY_FAIL;
```

Comment:
The `PyErr_Occurred` if below is no longer needed.

Reply:
Sorry below 'if' is setting the val pointer right. Please explain again.

Reply:
And the one below that `if` is checking `PyErr_Occurred()`. That's the one he's referring to.

Reply:
That said, it seems to me that the current code was already fine and did not need to be changed. We know the search shown in #24177 has false positives.

Reply:
It would be helpful to attempt write tests for each of these that fail in `main` before applying the fix. There are some true bugs that will be hard to expose in a test just because they are buried, but some of these false positives will be revealed by the failure to be able to write a test.

Reply:
FWIW, I think adding the exact test and removing `PyErr_Occurred()` is a clear improvement that would have been OK to include here since it follows the same pattern.  But no point in oscillating back and forth probably.